contracts/extensions/bridge-operator-governance/BridgeAdminOperator.sol

```diff
@@ -0,0 +1,423 @@
+
+  /**
+   * @inheritdoc IBridgeAdminOperator
+   */
+  function getTotalWeights() external view returns (uint256) {
+    return _totalWeight;
+  }
+
+  /**
+   * @inheritdoc IBridgeAdminOperator
+   */
+  function getBridgeVoterWeight(address governor) external view returns (uint256) {
+    return _getGovernorToBridgeOperatorInfo()[governor].voteWeight;
+  }
+
+  /**
+   * @inheritdoc IBridgeAdminOperator
+   */
+  function getBridgeVoterWeights(address[] calldata governors) external view returns (uint256[] memory weights) {
+    uint256 length = governors.length;
+    weights = new uint256[](length);
+    mapping(address => BridgeOperatorInfo) storage governorToBridgeOperatorInfo = _getGovernorToBridgeOperatorInfo();
+    for (uint256 i; i < length; ) {
+      weights[i] = governorToBridgeOperatorInfo[governors[i]].voteWeight;
+      unchecked {
+        ++i;
+      }
+    }
+  }
+
+  /**
+   * @inheritdoc IBridgeAdminOperator
+   */
+  function getSumBridgeVoterWeights(
+    address[] memory governors
+  ) public view nonDuplicate(governors) returns (uint256 sum) {
+    uint256 length = _getBridgeOperatorSet().length();
+    mapping(address => BridgeOperatorInfo) storage governorToBridgeOperatorInfo = _getGovernorToBridgeOperatorInfo();
+    for (uint256 i; i < length; ) {
+      sum += governorToBridgeOperatorInfo[governors[i]].voteWeight;
+
+      unchecked {
+        ++i;
+      }
+    }
+  }
+
+  /**
+   * @inheritdoc IBridgeAdminOperator
+   */
+  function totalBridgeOperators() external view returns (uint256) {
+    return _getBridgeOperatorSet().length();
+  }
+
+  /**
+   * @inheritdoc IBridgeAdminOperator
+   */
+  function isBridgeOperator(address addr) external view returns (bool) {
+    return _getBridgeOperatorSet().contains(addr);
+  }
+
+  /**
+   * @inheritdoc IBridgeAdminOperator
+   */
+  function getBridgeOperators() public view returns (address[] memory) {
+    return _getBridgeOperatorSet().values();
+  }
+
+  /**
+   * @inheritdoc IBridgeAdminOperator
+   */
+  function getGovernors() external view returns (address[] memory) {
+    return _getGovernorsSet().values();
+  }
+
+  /**
+   * @inheritdoc IBridgeAdminOperator
+   */
+  function getBridgeOperatorOf(address[] calldata governors) external view returns (address[] memory bridgeOperators_) {
+    uint256 length = governors.length;
+    bridgeOperators_ = new address[](length);
+
+    mapping(address => BridgeOperatorInfo) storage gorvernorToBridgeOperator = _getGovernorToBridgeOperatorInfo();
+    for (uint256 i; i < length; ) {
+      bridgeOperators_[i] = gorvernorToBridgeOperator[governors[i]].addr;
+      unchecked {
+        ++i;
+      }
+    }
+  }
+
+  /**
+   * @inheritdoc IQuorum
+   */
+  function minimumVoteWeight() public view virtual returns (uint256) {
+    return (_num * _totalWeight + _denom - 1) / _denom;
+  }
+
+  function _addBridgeOperators(
+    uint256[] memory voteWeights,
+    address[] memory governors,
+    address[] memory brigdeOperators
```

Comment:
Not sure if the arrays share the same length

Reply:
check provided.